spinalcordtoolbox/csa_pmj.py

```diff
@@ -44,16 +45,18 @@ def get_slices_for_pmj_distance(segmentation, pmj, distance, extent, param_cente
     _, _, Z = (im_seg_with_pmj.data > NEAR_ZERO_THRESHOLD).nonzero()
     min_z_index, max_z_index = min(Z), max(Z)
 
-    from spinalcordtoolbox.straightening import _get_centerline
     # Linear interpolation (vs. bspline) ensures strong robustness towards defective segmentations at the top slices.
     param_centerline.algo_fitting = 'linear'
     # On top of the linear interpolation we add some smoothing to remove discontinuities.
     param_centerline.smooth = 50
     param_centerline.minmax = True
-    # Compute spinalcordtoolbox.types.Centerline class
-    ctl_seg_with_pmj = _get_centerline(im_seg_with_pmj, param_centerline, verbose=verbose)
+    # Compute spinalcordtoolbox.types.Centerline class from get_centerline wit physical coordinates
+    _, arr_ctl_phys, arr_ctl_der_phys, _ = get_centerline(im_seg_with_pmj, param_centerline,
+                                                          verbose=verbose, space="phys")
+    ctl_seg_with_pmj = Centerline(points_x=arr_ctl_phys[0].tolist(), deriv_x=arr_ctl_der_phys[0].tolist(),
+                                  points_y=arr_ctl_phys[1].tolist(), deriv_y=arr_ctl_der_phys[1].tolist(),
+                                  points_z=arr_ctl_phys[2].tolist(), deriv_z=arr_ctl_der_phys[2].tolist())
```

Comment:
Maybe this is too tricky/obscure, but I believe the following shorter syntax would actually work:
```suggestion
    ctl_seg_with_pmj = Centerline(*arr_ctl_phys, *arr_ctl_der_phys)
```

To unpack this a little bit:
- You can use an iterable to supply positional arguments to a python function. This mirrors the syntax for defining a function which accepts a variable number of arguments:
  ```py3
  def my_function(a, b, c, *args): pass
  ```
  For function definitions, you can only have one `*args`, and it must be the last positional argument (so, keyword-only arguments can still go after), and `args` will be an actual `list`. But for function calls, you can put `*some_iterable` anywhere in the list, or even more than once, and it can be any iterable.
- For numpy arrays like `arr_ctl_phys` and `arr_ctl_der_phys`, iterating over them produces sub-arrays in the order `arr_ctl_phys[0], arr_ctl_phys[1], ...`, and the ordering happens to match the positional arguments to the `Centerline` constructor.
- This doesn't account for the `.tolist()` part, but the `Centerline` constructor happens to only use these parameters by iterating over them once. So they would act the same even without `.tolist()`.

But, again, maybe too obscure.

Reply:
!!! This actually works? This is lovely. 

I've used `*` syntax before when supplying arguments, but I hadn't realized it could be used on numpy arrays, too! 

Thanks for the tip. :hearts:

Reply:
Quick test:

![image](https://user-images.githubusercontent.com/16181459/220193235-ba53a4ac-1d2d-468b-b749-236cffc9479c.png)

![image](https://user-images.githubusercontent.com/16181459/220193304-8468c59d-82ac-42fd-abe1-115b3d64ce05.png)